logstash-core/lib/logstash/java_pipeline.rb

```diff
@@ -281,11 +281,7 @@ def resolve_cluster_uuids
 
   def wait_inputs
     @input_threads.each do |thread|
-      if thread.class == Java::JavaObject
-        thread.to_java.join
-      else
-        thread.join
-      end
+      thread.join # Thread or java.lang.Thread (both have #join)
```

Comment:
yeah, this is indeed weird. I wonder what the reasoning for using that was? looked in https://github.com/elastic/logstash/pull/10232 but did not see anything.

Reply:
it is related to the change here, [namely this one](https://github.com/elastic/logstash/pull/11391/files#diff-2d7b2300f0c481ab5779aeabb94c852dL68-R68)

Reply:
did you link the right change?

Reply:
thought so, since that is a place where inputs are `start`-ed and the returned 
`java.lang.Thread` used to be wrapped in a `JavaObject`